### ISSUE TITLE
Feature Request - CMYK Mode

### ISSUE THREAD

**Issue by avagp** [photopea/photopea#3702]
Hi Ivan.

I have had read a couple of times about CMYK mode on Photopea on the this github.

I work printing a lot of stuff, and for the years i have used Photopea, i solved this translation from sRGB to CMYK with other computer, but now this is getting me a lot of trouble.

I have had a couple of huge difference in color, and bad products in the end because of this incapacity in Photopea.

Here on the left i have the Photopea file. And on the right, after passing to Photoshop to CMYK color mode:
![image](https://user-images.githubusercontent.com/11801254/131767955-2d1c478a-111d-443b-9fde-197cf0878f79.png)

The difference is huge on brightness.

Can you please let us have a CMYK mode, like Photoshop does?

This is a really important feature, please

**Comment by MichalSZZ** [comment]
Yes I agree. That is "nice to have" in Photopea but I think that isn't so simple to implement. 
Look on GIMP, as I know - they still haven't real CMYK implementation.

When I added a book to print house, in their printing specyfication there was PDF file in CMYK.
I think they have so good printing machines and technicans and they could print even from RGB. But I had print sample of my page where i have one page as a 100% black. In sample there was photo changed to grayscale only. And effect wasn't good as in a real print where I chenged that to 100% K.

Adding CMYK to color pisker is very easy. But that still show RGB representation of that CMYK because monitor can't show some of CMYK color and CMYK can't have all colors from sRGB.

In Photoshop you can simulate CMYK view but it is good only when you have specific printer profile and know on what material it will be printed.

I think small group of Photopea users will use CMYK for the real work as you have.

**Comment by avagp** [comment]
I totally don't know if it is hard or don't, i am completely ignorant
about it.

At least for me, it is not a "nice to have" feature, it became a deal
breaker. I have to borrow another pc to finish my work, because Photopea
only, does not do the job.

I know CMYK changes on the monitor, but at least the difference gap
shrinks. As now, i have huge differences, which kills my work.

What do you think Ivan? Is it doable?  🤔

**Comment by avagp** [comment]
I have found a couple of sites that gives a patch solution to my issue.

https://www.rapidtables.com/convert/color/rgb-to-cmyk.html

Here you can see the maths behind it.

But here, it has more profiles and better correction.
https://www.rgb2cmyk.org/es.html

Here is what i got:
![image](https://user-images.githubusercontent.com/11801254/132264953-1ff4f01d-18f8-4806-962f-f9365c83a4d1.png)


Maybe you can contact them  :)

**Comment by photopea** [comment]
I know how to fix it, and I will fix it.

The problem is, that I have been trying to add the InDesign support for three weeks, with almost no progress, I am stuck. So I am thinking about giving up and going do other things. But on the other hand, I dont want to throw away three weeks of work :(

**Comment by MichalSZZ** [comment]
Maybe that CMYK support will be a great break for you and will clear up your mind. After that you can continue work with .indd support with new energy :)

**Comment by avagp** [comment]
I will have to ascribe to Michal here.

I also work creating, and a couple of times i have got frustrated, without an idea on how to proceed.

Let it go for a couple of days. Then come back refreshed. You will approach differently, I promise.

**Comment by avagp** [comment]
Any updates on this matter?

:)

**Comment by avagp** [comment]
Any progress on InDesign? Or in this issue?

**Comment by avagp** [comment]
Hi!

It have passed a month since the report, when do you plan to release this feature? 😢

**Comment by photopea** [comment]
I still have been looking into InDesign, without having any useful results (but I did some progress). I was solving minor issues in the meantime.

**Comment by photopea** [comment]
I have fixed an issue, which looked similar: https://github.com/photopea/photopea/issues/3667

What exactly is the problem with this issue? I thought it is abut a file, which looks different in Photopea than elsewhere, but you did not attach any files.

**Comment by avagp** [comment]
No, it is not.

I need Photopea to have a CYMK mode and a profile selector (if possible). Just like Photoshop does.

Here is a video explaining the problem, and how to solve it in Photoshop:
https://www.youtube.com/watch?v=8cK4dEBbHds

That is what i need.

And as i said, i work printing all my jobs, and some times (like the picture attached on the beginning) i have such a drop on color that the final product is too different from the original image, which causes me problems.

**Comment by avagp** [comment]
I believe what you describe is exactly how Photoshop works on CMYK mode.

:)

SWOP is a good standard as i have read. But it will be able to load profiles as well in the future?

**Comment by avagp** [comment]
I just wanted to say thanks!

It really has been a game changer. I use it on almost all my projects.

Thanks Ivan!

**Comment by photopea** [comment]
you are welcome :)
